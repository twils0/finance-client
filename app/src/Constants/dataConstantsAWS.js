// eslint-disable-next-line
export const actionTypes = {
  SET_AWS_STATUS: 'DATA_AUTH_AWS_STATUS',
  SET_AWS_USER: 'DATA_AUTH_SET_AWS_USER',
  RESET_AWS_STATE: 'DATA_AUTH_RESET_AWS_STATE',
};
