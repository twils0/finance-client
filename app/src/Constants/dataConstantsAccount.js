export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
