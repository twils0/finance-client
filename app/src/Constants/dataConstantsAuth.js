export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
