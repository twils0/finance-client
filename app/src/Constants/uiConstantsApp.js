export const actionTypes = {
  SET_FONTS_STATUS: 'APP_SET_FONTS_STATUS',
  LOAD_FONTS: 'APP_LOAD_FONTS',
  SET_IMAGE_STATUS: 'APP_SET_IMAGE_STATUS',
  LOAD_IMAGE: 'APP_LOAD_IMAGE',
  SET_STRIPE_STATUS: 'APP_SET_STRIPE_STATUS',
  SET_STRIPE_ELEMENT_LOADED: 'APP_SET_STRIPE_ELEMENT_LOADED',
  LOAD_STRIPE: 'APP_FETCH_STRIPE',
  SET_STRIPE_OBJECT: 'APP_SET_STRIPE_OBJECT',
};

export const motionStatusTypes = {
  IDLE: 'IDLE',
  ACTIVE: 'ACTIVE',
};

export const imageNames = {
  EXAMPLE_HEADER: 'exampleHeader',
};

export const errorMessages = {
  NO_EMAIL: 'Please enter your email address.',
  INVALID_EMAIL: 'Please enter a valid email address.',
  NO_PASSWORD: 'Please enter your password.',
  NO_NEW_PASSWORD: 'Please enter your new password.',
  INVALID_PASSWORD:
    'Please enter a password containing at least 8 characters, one of the following special characters, $@!%*#?&, one letter, and one number.',
  NO_MATCH_PASSWORD: 'This password does not match your password above.',
  NO_NAME_ON_CARD: 'Please enter the name shown on your credit card.',
  NO_STRIPE: 'Please enter your credit card information.',
  INVALID_CARD_NUMBER: 'Please enter a valid credit card number.',
  INCOMPLETE_CARD_NUMBER: 'Please enter your full credit card number.',
  INVALID_CARD_EXPIRATION_DATE: 'Please enter a valid expiration date.',
  INVALID_CARD_CVC: 'Please enter a valid CVC.',
  INVALID_CARD_ZIP_CODE: 'Please enter a valid zip code.',
  NO_NAME: 'Please enter your name.',
  NO_PHONE: 'Please enter your mobile phone number.',
  INVALID_PHONE: 'Please enter a valid mobile phone number.',
  NO_VERIFICATION_CODE_PHONE: 'Please enter the verification code sent to your mobile phone.',
  NO_VERIFICATION_CODE_EMAIL: 'Please enter the verification code sent to your email.',
  NO_CONFIRMATION_CODE_PHONE: 'Please enter the confirmation code sent to your mobile phone.',
  INVALID_VERIFICATION_CODE:
    'Please verify you have entered the correct verification code or request a new code.',
  INVALID_CONFIRMATION_CODE:
    'Please verify you have entered the correct confirmation code or request a new code.',
  EXPIRED_VERIFICATION_CODE: 'This verification code has expired.',
  EXPIRED_CONFIRMATION_CODE: 'This confirmation code has expired.',
  EMAIL_IN_USE:
    'An account using this email already exists. Please login or use a different email.',
  EMAIL_IN_USE_INTERNAL:
    'An account using this email already exists. Please use a different email.',
  EMAIL_NOT_FOUND:
    'This email is not associated with an account. Please enter another email or sign up.',
  PASSWORD_NOT_FOUND: 'This password does not match our records.',
  PROMO_CODE_INVALID: 'Please enter a valid promo code.',
  PROMO_CODE_EXPIRED: 'This promo code is no longer valid.',
  RETRY_SIGN_UP: 'Please click "Cancel" below and login.',
  INVALID_DELETE_ACCOUNT: 'Please type "delete" to confirm you would like to delete your account',
  LIMIT_EXCEEDED:
    'You have reached your login attempt limit. Please allow some time and try again.',
  INTERNAL_ERROR:
    'Please try your request again. If unsuccessful on subsequent attempts, please contact customer service.',
};
