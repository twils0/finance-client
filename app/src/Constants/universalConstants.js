import stripeConfig from '../../../stripe.config.json';
import dbConfig from '../../../db.config.json';
import iexConfig from '../../../iex.config.json';

export const baseURLs = {
  STRIPE: stripeConfig.url,
  DB: dbConfig.url,
  IEX: iexConfig.urls.api,
  IEX_DEV: iexConfig.urls.dev,
  IEX_TERMS: iexConfig.urls.terms,
};

export const URLs = {
  CUSTOMERS: '/customers',
  USERS: '/users',
  EMAILS: '/emails',
  SECURITIES: '/securities',
  SECURITIES_SEARCH: '/securities/search',
  IEX_COMPANY: tickerCusip => `/stock/${tickerCusip}/company`,
  IEX_QUOTE: tickerCusip => `/stock/${tickerCusip}/quote`,
};

export const axiosConfig = {
  STRIPE: {
    baseURL: baseURLs.STRIPE,
    timeout: 30000,
  },
  DB: {
    baseURL: baseURLs.DB,
    timeout: 30000,
  },
  IEX: {
    baseURL: baseURLs.IEX,
    timeout: 30000,
  },
};

export const pathNames = {
  DEFAULT: '*',
  LOGIN: '/login',
  SIGN_UP: '/signup',
  CODE: '/code',
  CODE_VERIFY_EMAIL: '/code/:verificationId?',
  TERMS: '/terms',
  INTERNAL: '/internal',
  WATCHLIST: '/internal/watchlist',
  WATCHLIST_SECURITY_ID: '/internal/watchlist/:securityId?',
  ACCOUNT: '/internal/account',
};

export const requestStatusTypes = {
  IDLE: 'IDLE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
