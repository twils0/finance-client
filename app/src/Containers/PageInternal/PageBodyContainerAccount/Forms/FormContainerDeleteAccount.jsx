import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { BeatLoader } from 'halogenium';

import theme from '../../../../themes';

import { requestStatusTypes } from '../../../../Constants/universalConstants';
import { statusNames } from '../../../../Constants/dataConstantsAuth';
import { errorMessages } from '../../../../Constants/uiConstantsApp';
import { formNames } from '../../../../Constants/uiConstantsAccount';
import { setInputValueError } from '../../../../Actions/uiActionsAccount';

import FlexColumn from '../../../../Components/Flex/FlexColumn';
import FormDeleteAccount from '../../../../Components/PageInternal/PageBodyAccount/Forms/FormDeleteAccount';

class FormContainerDeleteAccount extends React.Component {
  componentDidMount() {
    this.props.setHeight(formNames.DELETE_ACCOUNT);
  }

  componentWillUpdate(nextProps) {
    if (nextProps.statusAuth[statusNames.DELETE_ACCOUNT].status === requestStatusTypes.LOADING) {
      nextProps.clearElement(formNames.DELETE_ACCOUNT);
    }
  }

  componentDidUpdate() {
    this.props.setHeight(formNames.DELETE_ACCOUNT);
  }

  componentWillUnmount() {
    this.props.clearElement(formNames.DELETE_ACCOUNT);
  }

  handleInputChange = (inputEvent) => {
    inputEvent.preventDefault();
    const { inputs, handleInputValueError } = this.props;
    const input = { ...inputs[inputEvent.target.id] };

    if (input) {
      if (inputEvent.target.value || input.value) {
        input.value = inputEvent.target.value;
      }

      if (input.errorMessage) {
        switch (input.errorMessage) {
          case errorMessages.INVALID_DELETE_ACCOUNT: {
            if (input.value && input.value === 'delete') {
              input.errorMessage = '';
            }
            break;
          }
          default:
            input.errorMessage = '';
            break;
        }
      }
    }

    handleInputValueError(input);
  };

  render() {
    const {
      heightRef,
      rowWidthIndent,
      headerWidthBufferRight,
      mediaWidth,
      height,
      rowHeight,
      rowHeightBuffer,
      statusAuth,
      currentForm,
      editForm,
      inputs,
    } = this.props;

    if (statusAuth[statusNames.DELETE_ACCOUNT].status === requestStatusTypes.LOADING) {
      const heightString = `${height}px`;

      return (
        <FlexColumn width="100%" height={heightString}>
          <BeatLoader color={theme.color.secondary.default} />
        </FlexColumn>
      );
    }
    return (
      <FormDeleteAccount
        heightRef={heightRef}
        rowWidthIndent={rowWidthIndent}
        headerWidthBufferRight={headerWidthBufferRight}
        mediaWidth={mediaWidth}
        rowHeight={rowHeight}
        rowHeightBuffer={rowHeightBuffer}
        currentFormBool={currentForm === formNames.DELETE_ACCOUNT}
        editForm={editForm}
        inputs={inputs}
        handleInputChange={this.handleInputChange}
      />
    );
  }
}

FormContainerDeleteAccount.propTypes = {
  heightRef: PropTypes.func.isRequired,
  rowWidthIndent: PropTypes.number.isRequired,
  headerWidthBufferRight: PropTypes.number.isRequired,
  mediaWidth: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  rowHeight: PropTypes.number.isRequired,
  rowHeightBuffer: PropTypes.number.isRequired,
  statusAuth: PropTypes.object.isRequired,
  currentForm: PropTypes.string.isRequired,
  editForm: PropTypes.bool.isRequired,
  inputs: PropTypes.object.isRequired,
  setHeight: PropTypes.func.isRequired,
  clearElement: PropTypes.func.isRequired,
  handleInputValueError: PropTypes.func.isRequired,
};

FormContainerDeleteAccount.defaultProps = {};

const mapStateToProps = (state, ownProps) => ({
  heightRef: ownProps.heightRef,
  rowWidthIndent: state.ui.internal.account.utilities.forms.rowWidthIndent,
  headerWidthBufferRight: state.ui.internal.account.utilities.forms.headerWidthBufferRight,
  mediaWidth: state.ui.internal.account.utilities.pageBody.mediaWidth,
  height: state.ui.internal.account.forms[formNames.BILLING].height,
  rowHeight: state.ui.internal.account.utilities.forms.rowHeight,
  rowHeightBuffer: state.ui.internal.account.utilities.forms.rowHeightBuffer,
  heightBuffer: state.ui.internal.account.utilities.forms.heightBuffer,
  currentForm: state.ui.internal.account.forms.current,
  statusAuth: state.data.auth.status,
  editForm: state.ui.internal.account.forms[formNames.DELETE_ACCOUNT].edit,
  inputs: state.ui.internal.account.forms[formNames.DELETE_ACCOUNT].inputs,
  setHeight: ownProps.setHeight,
  clearElement: ownProps.clearElement,
});

const mapDispatchToProps = dispatch => ({
  handleInputValueError: payload => dispatch(setInputValueError(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(FormContainerDeleteAccount);
