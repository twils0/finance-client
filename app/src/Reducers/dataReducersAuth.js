import { combineReducers } from 'redux';

import { requestStatusTypes, pathNames } from '../Constants/universalConstants';
import { actionTypes, statusNames, codeTypeNames } from '../Constants/dataConstantsAuth';

export const initialState = {
  authenticated: false,
  redirectURL: pathNames.WATCHLIST,
  status: {
    [statusNames.LOGIN]: {
      id: statusNames.LOGIN,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_OUT_DEVICES]: {
      id: statusNames.SIGN_OUT_DEVICES,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGIN_MFA]: {
      id: statusNames.LOGIN_MFA,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_UP]: {
      id: statusNames.SIGN_UP,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE]: {
      id: statusNames.VERIFY_PHONE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE_CODE]: {
      id: statusNames.VERIFY_PHONE_CODE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL]: {
      id: statusNames.VERIFY_EMAIL,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL_LINK]: {
      id: statusNames.VERIFY_EMAIL_LINK,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.FORGOT_PASSWORD]: {
      id: statusNames.FORGOT_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.RESET_PASSWORD]: {
      id: statusNames.RESET_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.CHANGE_PASSWORD]: {
      id: statusNames.CHANGE_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGOUT]: {
      id: statusNames.LOGOUT,
      status: requestStatusTypes.SUCCESS,
    },
    [statusNames.DELETE_ACCOUNT]: {
      id: statusNames.DELETE_ACCOUNT,
      status: requestStatusTypes.SUCCESS,
    },
  },
  codeTypes: {
    [codeTypeNames.VERIFY_PHONE]: {
      id: codeTypeNames.VERIFY_PHONE,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL]: {
      id: codeTypeNames.VERIFY_EMAIL,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL_ADDITIONAL]: {
      id: codeTypeNames.VERIFY_EMAIL_ADDITIONAL,
      needed: true,
    },
  },
};

export const authenticated = (state = initialState.authenticated, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTHENTICATED:
      return action.payload.authenticated;
    case actionTypes.RESET_AUTH_STATE:
      return initialState.authenticated;
    default:
      return state;
  }
};

export const redirectURL = (state = initialState.redirectURL, action) => {
  switch (action.type) {
    case actionTypes.SET_REDIRECT_URL:
      return action.payload.redirectURL;
    default:
      return state;
  }
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTH_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const codeTypes = (state = initialState.codeTypes, action) => {
  switch (action.type) {
    case actionTypes.SET_CODE_TYPE:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.codeTypes;
    default:
      return state;
  }
};

const dataReducersAuth = combineReducers({
  authenticated,
  redirectURL,
  status,
  codeTypes,
});

export default dataReducersAuth;
