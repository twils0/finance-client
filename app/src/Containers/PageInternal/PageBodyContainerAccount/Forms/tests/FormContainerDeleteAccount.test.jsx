import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import { initialState as initialStateAuth } from '../../../../../Reducers/dataReducersAuth';
import { initialState as initialStateUIAccount } from '../../../../../Reducers/uiReducersAccount';
import { requestStatusTypes } from '../../../../../Constants/universalConstants';
import { statusNames } from '../../../../../Constants/dataConstantsAuth';
import { errorMessages } from '../../../../../Constants/uiConstantsApp';
import {
  actionTypes as actionTypesUIAccount,
  formNames,
  inputNames,
} from '../../../../../Constants/uiConstantsAccount';

import FormContainerDeleteAccount from '../FormContainerDeleteAccount';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const heightRef = jest.fn();
const setHeight = jest.fn();
const clearElement = jest.fn();

const deleteValue = 'delete';

const shallowComponent = store =>
  shallow(
    <FormContainerDeleteAccount
      heightRef={heightRef}
      setHeight={setHeight}
      clearElement={clearElement}
    />,
    {
      context: {
        store,
      },
    },
  ).dive();

describe('Containers', () => {
  describe('PageInternal', () => {
    describe('PageBodyContainerAccount', () => {
      describe('Forms', () => {
        describe('FormContainerDeleteAccount', () => {
          afterEach(() => {
            heightRef.mockReset();
            setHeight.mockReset();
            clearElement.mockReset();
          });

          it('shallow renders correctly', async () => {
            const stateBeforeAuth = JSON.parse(JSON.stringify(initialStateAuth));
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));

            const store = mockStore({
              data: {
                auth: stateBeforeAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const expectedActions = [];

            const wrapper = shallowComponent(store);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
            expect(setHeight).toBeCalledWith(formNames.DELETE_ACCOUNT);
          });

          it('shallow renders correctly, with componentWillUpdate, componentDidUpdate, and componentWillUnmount', async () => {
            const stateBeforeAuth = JSON.parse(JSON.stringify(initialStateAuth));
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));
            const { forms } = stateBeforeUIAccount;

            forms[formNames.DELETE_ACCOUNT].inputs[
              inputNames[formNames.DELETE_ACCOUNT].CONFIRM
            ].value = deleteValue;

            const store = mockStore({
              data: {
                auth: stateBeforeAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const nextProps = {
              statusAuth: {
                [statusNames.DELETE_ACCOUNT]: {
                  status: requestStatusTypes.LOADING,
                },
              },
              clearElement,
            };

            const expectedActions = [];

            const wrapper = shallowComponent(store);

            const instance = wrapper.instance();

            instance.componentWillUpdate(nextProps);
            instance.componentDidUpdate();
            instance.componentWillUnmount();

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
            expect(setHeight).toBeCalledWith(formNames.DELETE_ACCOUNT);
            expect(clearElement).toBeCalledWith(formNames.DELETE_ACCOUNT);
            expect(setHeight).toBeCalledWith(formNames.DELETE_ACCOUNT);
            expect(clearElement).toBeCalledWith(formNames.DELETE_ACCOUNT);
          });

          it('shallow renders correctly, with loading indicator', async () => {
            const stateBeforeAuth = JSON.parse(JSON.stringify(initialStateAuth));
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));

            stateBeforeAuth.status[statusNames.DELETE_ACCOUNT].status = requestStatusTypes.LOADING;

            const store = mockStore({
              data: {
                auth: stateBeforeAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const expectedActions = [];

            const wrapper = shallowComponent(store);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
          });

          it('shallow renders correctly and calls handleInputChange, with invalid delete account', async () => {
            const stateBeforeAuth = JSON.parse(JSON.stringify(initialStateAuth));
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));
            const { forms } = stateBeforeUIAccount;

            const wrongDeleteValue = 'del';

            forms[formNames.DELETE_ACCOUNT].inputs[
              inputNames[formNames.DELETE_ACCOUNT].CONFIRM
            ].value = wrongDeleteValue;
            forms[formNames.DELETE_ACCOUNT].inputs[
              inputNames[formNames.DELETE_ACCOUNT].CONFIRM
            ].errorMessage =
              errorMessages.INVALID_DELETE_ACCOUNT;

            const store = mockStore({
              data: {
                auth: stateBeforeAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const uiDeleteAccountConfirm = JSON.parse(
              JSON.stringify(
                forms[formNames.DELETE_ACCOUNT].inputs[
                  inputNames[formNames.DELETE_ACCOUNT].CONFIRM
                ],
              ),
            );

            uiDeleteAccountConfirm.value = deleteValue;
            uiDeleteAccountConfirm.errorMessage = '';

            const inputEvent = {
              preventDefault: jest.fn(),
              target: {
                id: inputNames[formNames.DELETE_ACCOUNT].CONFIRM,
                value: deleteValue,
              },
            };
            const expectedActions = [
              {
                type: actionTypesUIAccount.SET_INPUT_VALUE_ERROR,
                payload: uiDeleteAccountConfirm,
              },
            ];

            const wrapper = shallowComponent(store);

            const instance = wrapper.instance();

            instance.handleInputChange(inputEvent);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
          });

          it('shallow renders correctly and calls handleInputChange, with undefined error', async () => {
            const stateBeforeAuth = JSON.parse(JSON.stringify(initialStateAuth));
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));
            const { forms } = stateBeforeUIAccount;

            forms[formNames.DELETE_ACCOUNT].inputs[
              inputNames[formNames.DELETE_ACCOUNT].CONFIRM
            ].errorMessage =
              errorMessages.INTERNAL_ERROR;

            const store = mockStore({
              data: {
                auth: stateBeforeAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const uiDeleteAccountConfirm = JSON.parse(
              JSON.stringify(
                forms[formNames.DELETE_ACCOUNT].inputs[
                  inputNames[formNames.DELETE_ACCOUNT].CONFIRM
                ],
              ),
            );

            uiDeleteAccountConfirm.value = deleteValue;
            uiDeleteAccountConfirm.errorMessage = '';

            const inputEvent = {
              preventDefault: jest.fn(),
              target: {
                id: inputNames[formNames.DELETE_ACCOUNT].CONFIRM,
                value: deleteValue,
              },
            };
            const expectedActions = [
              {
                type: actionTypesUIAccount.SET_INPUT_VALUE_ERROR,
                payload: uiDeleteAccountConfirm,
              },
            ];

            const wrapper = shallowComponent(store);

            const instance = wrapper.instance();

            instance.handleInputChange(inputEvent);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
          });
        });
      });
    });
  });
});
