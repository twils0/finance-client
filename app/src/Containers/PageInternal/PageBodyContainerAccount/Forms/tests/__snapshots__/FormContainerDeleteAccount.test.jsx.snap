// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerDeleteAccount shallow renders correctly 1`] = `
<FormDeleteAccount
  currentFormBool={false}
  editForm={false}
  handleInputChange={[Function]}
  headerWidthBufferRight={10}
  heightRef={[MockFunction]}
  inputs={
    Object {
      "deleteAccountConfirm": Object {
        "errorMessage": "",
        "id": "deleteAccountConfirm",
        "parentId": "deleteAccountForm",
        "value": "",
      },
    }
  }
  mediaWidth={874}
  rowHeight={60}
  rowHeightBuffer={18}
  rowWidthIndent={10}
/>
`;

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerDeleteAccount shallow renders correctly and calls handleInputChange, with invalid delete account 1`] = `
<FormDeleteAccount
  currentFormBool={false}
  editForm={false}
  handleInputChange={[Function]}
  headerWidthBufferRight={10}
  heightRef={[MockFunction]}
  inputs={
    Object {
      "deleteAccountConfirm": Object {
        "errorMessage": "Please type \\"delete\\" to confirm you would like to delete your account",
        "id": "deleteAccountConfirm",
        "parentId": "deleteAccountForm",
        "value": "del",
      },
    }
  }
  mediaWidth={874}
  rowHeight={60}
  rowHeightBuffer={18}
  rowWidthIndent={10}
/>
`;

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerDeleteAccount shallow renders correctly and calls handleInputChange, with undefined error 1`] = `
<FormDeleteAccount
  currentFormBool={false}
  editForm={false}
  handleInputChange={[Function]}
  headerWidthBufferRight={10}
  heightRef={[MockFunction]}
  inputs={
    Object {
      "deleteAccountConfirm": Object {
        "errorMessage": "Please try your request again. If unsuccessful on subsequent attempts, please contact customer service.",
        "id": "deleteAccountConfirm",
        "parentId": "deleteAccountForm",
        "value": "",
      },
    }
  }
  mediaWidth={874}
  rowHeight={60}
  rowHeightBuffer={18}
  rowWidthIndent={10}
/>
`;

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerDeleteAccount shallow renders correctly, with componentWillUpdate, componentDidUpdate, and componentWillUnmount 1`] = `
<FormDeleteAccount
  currentFormBool={false}
  editForm={false}
  handleInputChange={[Function]}
  headerWidthBufferRight={10}
  heightRef={[MockFunction]}
  inputs={
    Object {
      "deleteAccountConfirm": Object {
        "errorMessage": "",
        "id": "deleteAccountConfirm",
        "parentId": "deleteAccountForm",
        "value": "delete",
      },
    }
  }
  mediaWidth={874}
  rowHeight={60}
  rowHeightBuffer={18}
  rowWidthIndent={10}
/>
`;

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerDeleteAccount shallow renders correctly, with loading indicator 1`] = `
<FlexColumn
  alignItems={null}
  alignSelf={null}
  borderColorType="border"
  borderRadius={null}
  borderStyle={null}
  borderWidth={null}
  boxSizing={null}
  color={Object {}}
  display={null}
  flex={null}
  flexWrap={null}
  height="0px"
  justifyContent={null}
  margin={null}
  mediaWidth={null}
  minHeight={null}
  minWidth={null}
  overflowX={null}
  overflowY={null}
  padding={null}
  position={null}
  responsiveAlignSecurities={null}
  responsiveJustifyContent={null}
  responsiveMargin={null}
  responsivePadding={null}
  responsivePosition={null}
  responsiveWidth={null}
  separator={false}
  themeColor="primary"
  width="100%"
>
  <BeatLoader
    color="#7367e6"
    loading={true}
    margin="2px"
    size="15px"
  />
</FlexColumn>
`;
